File: backend/agentic_crm_sdk/agents.py
"""
Agents API client module.
"""
from agentic_crm_sdk.client import AgenticCRM


class AgentsClient:
    def __init__(self, client: AgenticCRM):
        self._client = client

    def list(self, **params):
        return self._client.get("/api/agents", params=params)

    def get(self, agent_id: int):
        return self._client.get(f"/api/agents/{agent_id}")

    def update_status(self, agent_id: int, status: str):
        return self._client.patch(f"/api/agents/{agent_id}", json={"status": status})

    def pause(self, agent_id: int):
        return self.update_status(agent_id, "paused")

    def resume(self, agent_id: int):
        return self.update_status(agent_id, "active")

File: frontend/lib/sdk/agents.ts
import type { Agent, AgentStatus } from '../api'
import { AgenticCRM } from './client'

export interface CreateAgentInput {
  name: string
  role: string
  description?: string
  config?: Record<string, unknown>
}

export interface UpdateAgentInput {
  name?: string
  role?: string
  description?: string
  config?: Record<string, unknown>
}

export class AgentsClient {
  constructor(private client: AgenticCRM) {}

  async list(): Promise<Agent[]> {
    return this.client.get<Agent[]>('/api/agents')
  }

  async get(agentId: number): Promise<Agent> {
    return this.client.get<Agent>(`/api/agents/${agentId}`)
  }

  async create(data: CreateAgentInput): Promise<Agent> {
    return this.client.post<Agent>('/api/agents', data)
  }

  async update(agentId: number, data: UpdateAgentInput): Promise<Agent> {
    return this.client.patch<Agent>(`/api/agents/${agentId}`, data)
  }

  async updateStatus(agentId: number, status: AgentStatus): Promise<Agent> {
    return this.client.patch<Agent>(`/api/agents/${agentId}`, { status })
  }

  async pause(agentId: number): Promise<Agent> {
    return this.updateStatus(agentId, 'paused')
  }

  async resume(agentId: number): Promise<Agent> {
    return this.updateStatus(agentId, 'active')
  }
}